### PR TITLE
Working todo component

### PR DESCRIPTION
**Pull request by sashkivna**

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+var app = angular.module('MyApp', [
```
Comment (r0ss0neri) [inline app/app.js]: eliminate `var app =`

File: app/app.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    'todo.module'
+]);
+
+app.controller('MyController', ['$scope', function ($scope) {
```
Comment (r0ss0neri) [inline app/app.js]: create separate file for controller `MyController `

File: app/index.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html ng-app="MyApp">
+<head>
+    <title></title>
+    <link rel="stylesheet" type="text/css"
+          href="http://netdna.bootstrapcdn.com/twitter-bootstrap/2.2.2/css/bootstrap-combined.min.css"/>
+    <link rel="stylesheet" href="styles.css">
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
```
Comment (r0ss0neri) [inline app/index.html]: add jqery to `package.json` as dependency

File: app/index.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+</head>
+<body id="body" ng-controller="MyController">
```
Comment (r0ss0neri) [inline app/index.html]: do you need `id` for `body`?

File: app/todo/todo.directive.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+angular.module('todo.module')
+    .directive('todo', function () {
+        return {
+            restrict: 'E',
+            templateUrl: 'toDo/toDo.template.html',
```
Comment (r0ss0neri) [inline app/todo/todo.directive.js]: blank line

File: app/todo/todo.directive.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            scope: {
+                tasks: '='
+            },
+
+            link: function ($scope) {
+
```
Comment (r0ss0neri) [inline app/todo/todo.directive.js]: remove blank line

File: app/todo/todo.directive.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+                $scope.tasks = [];
```
Comment (r0ss0neri) [inline app/todo/todo.directive.js]: add blank line

File: app/todo/todo.directive.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+                $scope.add = function (title) {
+                        if(title === undefined || title.length === 0) {
+                            alert("You tried to add empty task, pls type some description");
+                        }
```
Comment (r0ss0neri) [inline app/todo/todo.directive.js]: move `else` to prev line

File: app/todo/todo.directive.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+            templateUrl: 'toDo/toDo.template.html',
+            scope: {
+                tasks: '='
+            },
+
+            link: function ($scope) {
+
+                $scope.tasks = [];
+                $scope.add = function (title) {
+                        if(title === undefined || title.length === 0) {
```
Comment (r0ss0neri) [inline app/todo/todo.directive.js]: if (title) {
        $scope.tasks.push({task: $scope.title});
        $scope.title = "";
    }

File: app/todo/todo.directive.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+                        else {
+                            $scope.tasks.push({task: $scope.title});
+                            $scope.title = "";
+                        }
+                };
+
+                $scope.delete = function () {
+                    $scope.tasks.splice(this.$index, 1);
+                }
+
```
Comment (r0ss0neri) [inline app/todo/todo.directive.js]: remove blank line

File: app/todo/todo.spec.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
+/*
```
Comment (r0ss0neri) [inline app/todo/todo.spec.js]: remove all commented code

File: app/todo/todo.template.html [hunk 12]
```diff
@@ -0,0 +1,7 @@
+<form ng-submit="add(title)">
+    <input ng-model="title">
+</form>
+<div ng-repeat="todo in tasks track by $index">
+    <input type='checkbox' ng-model='task.done'/>
```
Comment (r0ss0neri) [inline app/todo/todo.template.html]: use `""` in template instead of `''`

File: karma.conf.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
+// Karma configuration
+// Generated on Thu Sep 27 2018 10:55:00 GMT+0300 (Финляндия (лето))
```
Comment (r0ss0neri) [inline karma.conf.js]: remove this

File: karma.conf.js [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+        // frameworks to use
+        // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+        frameworks: ['jasmine'],
+
+        // list of files / patterns to load in the browser
+        files: [
+            '../node_modules/jquery/dist/jquery.js',
+            '../node_modules/angular/angular.js',
+            '../node_modules/angular-mocks/angular-mocks.js',
+         /*   './app.js',
```
Comment (r0ss0neri) [inline karma.conf.js]: remove

**Review by r0ss0neri — CHANGES_REQUESTED** [review]

File: app/app.js (hunk 1 above)
Comment (r0ss0neri) [reply app/app.js]: rename `app.js` to `my-app.module.js` and module to 'my-app.module'

**Comment by r0ss0neri** [comment]
create folder components und move `todo` folder there

File: app/app.js (hunk 1 above)
Comment (sashkivna) [reply app/app.js]: done

File: app/app.js (hunk 2 above)
Comment (sashkivna) [reply app/app.js]: done

File: app/index.html (hunk 3 above)
Comment (sashkivna) [reply app/index.html]: done

File: app/index.html (hunk 4 above)
Comment (sashkivna) [reply app/index.html]: deleted id, done

File: app/todo/todo.directive.js (hunk 5 above)
Comment (sashkivna) [reply app/todo/todo.directive.js]: done

File: app/todo/todo.directive.js (hunk 10 above)
Comment (sashkivna) [reply app/todo/todo.directive.js]: done

File: app/todo/todo.spec.js (hunk 11 above)
Comment (sashkivna) [reply app/todo/todo.spec.js]: done

File: app/todo/todo.template.html (hunk 12 above)
Comment (sashkivna) [reply app/todo/todo.template.html]: done

File: karma.conf.js (hunk 13 above)
Comment (sashkivna) [reply karma.conf.js]: done

File: karma.conf.js (hunk 14 above)
Comment (sashkivna) [reply karma.conf.js]: done

File: app/app.js [hunk 15]
```diff
@@ -0,0 +1,4 @@
+    angular.module('MyApp', [
+        'todo.module'
+    ]);
+
```
Comment (r0ss0neri) [inline app/app.js]: remove blank line

File: app/app.controller.js [hunk 16]
```diff
@@ -0,0 +1,6 @@
+angular.module('MyApp', [
+    'todo.module'
+]).controller('MyController', ['$scope', function ($scope) {
+
+}]);
+
```
Comment (r0ss0neri) [inline app/app.controller.js]: remove blank line

File: app/app.controller.js [hunk 17]
```diff
@@ -0,0 +1,6 @@
+angular.module('MyApp', [
```
Comment (r0ss0neri) [inline app/app.controller.js]: `angular.module('MyApp').controller(...`

**Review by r0ss0neri — CHANGES_REQUESTED** [review]

File: app/app.controller.js [hunk 18]
```diff
@@ -0,0 +1,3 @@
+angular.module('MyApp')
```
Comment (r0ss0neri) [inline app/app.controller.js:1]: angular.module('MyApp')
        .controller('MyController', function ($scope) {
    });

**Review by r0ss0neri — CHANGES_REQUESTED** [review]

**Pull request merged**